extensions/http/jetty/src/main/java/org/eclipse/dataspaceconnector/extension/jetty/JettyExtension.java

```diff
@@ -39,11 +51,21 @@ public String name() {
     @Override
     public void initialize(ServiceExtensionContext context) {
         var monitor = context.getMonitor();
+        KeyStore ks = null;
+        var keystorePath = context.getConfig().getString(KEYSTORE_PATH_SETTING, null);
+        var configuration = JettyConfiguration.createFromConfig(context.getSetting(KEYSTORE_PASSWORD, "password"), context.getSetting(KEYMANAGER_PASSWORD, "password"), context.getConfig());
+
+        if (keystorePath != null) {
+            try {
+                ks = KeyStore.getInstance(context.getSetting(KEYSTORE_TYPE_SETTING, "PKCS12"));
+                ks.load(new FileInputStream(keystorePath), configuration.getKeystorePassword().toCharArray());
```

Comment:
try-with-resource possible here?

Reply:
KeyStore is not an `AutoClosable`, so we'd have to wrap that in a conventional `try-catch` anyway. Having another, inner `try-with-resources`, just for the file input stream seams a bit convoluted